crates/store/re_video/src/time.rs

```diff
@@ -28,50 +28,67 @@ impl Time {
         Self(v)
     }
 
+    /// `time_base` specifies the
     #[inline]
-    pub fn from_secs(v: f64, timescale: Timescale) -> Self {
-        Self((v * timescale.0 as f64).round() as i64)
+    pub fn from_secs_since_start(
+        secs_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self((secs_since_start * timescale.0 as f64).round() as i64 + start_time.0)
     }
 
     #[inline]
-    pub fn from_millis(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e3, timescale)
+    pub fn from_millis_since_start(
+        millis_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(millis_since_start / 1e3, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_micros(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e6, timescale)
+    pub fn from_micros_since_start(
+        micros_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(micros_since_start / 1e6, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_nanos(v: i64, timescale: Timescale) -> Self {
-        Self::from_secs(v as f64 / 1e9, timescale)
+    pub fn from_nanos_since_start(
+        nanos_since_start: i64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(nanos_since_start as f64 / 1e9, timescale, start_time)
     }
 
     /// Convert to a duration
     #[inline]
     pub fn duration(self, timescale: Timescale) -> std::time::Duration {
-        std::time::Duration::from_nanos(self.into_nanos(timescale) as _)
+        std::time::Duration::from_nanos(self.into_nanos_since_start(timescale, Self(0)) as _)
     }
 
     #[inline]
-    pub fn into_secs(self, timescale: Timescale) -> f64 {
-        self.0 as f64 / timescale.0 as f64
+    pub fn into_secs_since_start(self, timescale: Timescale, start_time: Self) -> f64 {
+        (self.0 - start_time.0) as f64 / timescale.0 as f64
     }
 
     #[inline]
-    pub fn into_millis(self, timescale: Timescale) -> f64 {
-        self.into_secs(timescale) * 1e3
+    pub fn into_millis_since_start(self, timescale: Timescale, start_time: Self) -> f64 {
+        self.into_secs_since_start(timescale, start_time) * 1e3
     }
 
     #[inline]
-    pub fn into_micros(self, timescale: Timescale) -> f64 {
-        self.into_secs(timescale) * 1e6
+    pub fn into_micros_since_start(self, timescale: Timescale, start_time: Self) -> f64 {
+        self.into_secs_since_start(timescale, start_time) * 1e6
     }
 
     #[inline]
-    pub fn into_nanos(self, timescale: Timescale) -> i64 {
-        (self.into_secs(timescale) * 1e9).round() as i64
+    pub fn into_nanos_since_start(self, timescale: Timescale, start_time: Self) -> i64 {
```

Comment:
Same here; these could be `into_duration_since_start(self, …) -> std::time::Duration`